activestorage/app/models/active_storage/blob.rb

```diff
@@ -350,7 +350,8 @@ def compute_checksum_in_chunks(io)
       raise ArgumentError, "io must be rewindable" unless io.respond_to?(:rewind)
 
       OpenSSL::Digest::MD5.new.tap do |checksum|
-        while chunk = io.read(5.megabytes)
+        buffer = +""
+        while chunk = io.read(5.megabytes, buffer)
           checksum << chunk
         end
```

Comment:
Could also write this as

```suggestion
        while io.read(5.megabytes, buffer)
          checksum << buffer
        end
```

Though I don't have a strong preference.

Reply:
seems reasonable. For clarity, I have renamed `buffer` to `read_buffer`.